File: atcoder/abc/abc156/c/main.py
def main():
    N = int(input())
    X = [int(i) for i in input().split()]

    ans = 10**9
    for pos in range(101):
        dist = 0
        for x in X:
            dist += (pos-x)**2
        ans = min(ans, dist)
    print(ans)
main()
            

File: atcoder/abc/abc156/c/main.cpp
#include<bits/stdc++.h>
using namespace std;


int main(){
  int N;
  cin >> N;
  int ans = 1e9;
  vector<int> X(N);
  for(int i=0;i<N;i++){
    cin >> X[i];
  }
  for(int xx=0;xx<150;xx++){
    int memo = 0;
    for(auto x: X){
      memo += (x-xx)*(x-xx);
    }
    ans = min(ans, memo);
  }
  cout << ans << endl;
}